pkg/tfbridge/x/tokens.go

```diff
@@ -649,62 +649,96 @@ func aliasResource(
 	if res == nil {
 		return
 	}
-
-	safeRange := func(r shim.Resource, f func(string, shim.Schema)) {
+	var walk func(*fieldHistory, *b.SchemaInfo, shim.Schema) (bool, bool)
+	walkResource := func(r shim.Resource, hist *map[string]*fieldHistory, info *map[string]*b.SchemaInfo) (bool, bool) {
 		if r == nil {
-			return
+			return hist != nil, info != nil
 		}
 		m := r.Schema()
 		if m == nil {
-			return
+			return hist != nil, info != nil
 		}
+
+		var rHasH, rHasI bool
+
 		m.Range(func(k string, v shim.Schema) bool {
-			f(k, v)
+			h, hasH := getNonNil(hist, k)
+			i, hasI := getNonNil(info, k)
+			keepH, keepI := walk(h, i, v)
+
+			hasH = hasH || keepH
+			hasI = hasI || keepI
+
+			if !hasH {
+				delete(*hist, k)
```

Comment:
This deletes both the type/field key and its content, correct?

Reply:
This removes the map entry indexed by the key `k` (`string) from the map `*hist` (`map[string]*fieldHistory`).

Did that answer your question?